students/AndrewMiotke/session02/grid_printer.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python3
+
+# Long hand way of creating a grid
+# def verbose_grid():
+#     print("+",  " - " * 4, "+", " - " * 4, "+")
+
+#     print("|              |              |")
+#     print("|              |              |")
+#     print("|              |              |")
+#     print("|              |              |")
+
+#     print("+",  " - " * 4, "+", " - " * 4, "+")
+
+#     print("|              |              |")
+#     print("|              |              |")
+#     print("|              |              |")
+#     print("|              |              |")
+
+#     print("+",  " - " * 4, "+", " - " * 4, "+")
+
+# verbose_grid()
+
+# More general grid generator
+def create_beam(beams):
+    beam = '+ - - - - '
+    plus = "+"
+    end_line = end=' '
+
+    print(beam * beams + plus + end_line)
+
+def create_post(posts):
+    post = '|         '
+
+    print(post * (posts + 1))
+
+def create_grid(beams, posts):
+    """
+    Returns a grid based on the parameters of this function
+    :param arg 2: define the number of beams(x axis) required to complete a cell
+    :param arg 3: define the number of posts(y axis) required to complete a cell
+
+    Example: create_grid(10, 10) would create a 10x10 grid
+    """
+    for _ in range(posts):
+        create_beam(beams)
```

Comment:
This is a good start but do note that you're hardcoding the size of each cell. 

The initial print_grid should always create a grid of 4 cells with the total height/length being the parameter we give it. For example: print_grid(9) should return a 4x4 grid where the entire length/height (2 cells worth) is 9. Hint: The middle '+' will handle it such that the two cells are of 4 length each. 

The final print_grid should take two parameters.  The first parameter determines the nxn grid size and the second parameter determines the size of each cell. So for example, print_grid(5,3) should return a 5x5 grid where each cell is 3x3 (3 '-' across and 3 '|' long )

Reply:
Thank you for the feedback. I ended up writing a new file and I believe it's working now. Should be in the latest commit.